groups/nts/ntsu/ntsu_timestamputil.h

```diff
@@ -19,49 +19,120 @@
 #include <bsls_ident.h>
 BSLS_IDENT("$Id: $")
 
+#include <ntscfg_platform.h>
+#include <ntsscm_version.h>
+#include <bsl_cstddef.h>
+
 namespace BloombergLP {
 namespace ntsu {
 
-/// This struct is used to define/redefine types and constants used for Linux
-/// network timestamping feature.
+/// @internal @brief 
+/// Redefine types and constants used for Linux network timestamping in a 
+/// portable way, independant of any operating system headers and content of
+/// those headers that may or may not be present on the build machine.
+///
+/// @details
+/// The following caveats apply to timestamping support:
+///
+/// @li Timestamping is only supported for Linux kernel versions >= 4.18.0.
+///
+/// @li Timestamping is only supported for TCP and UDP sockets.
+///
+/// @li Outgoing timestamping may only be enabled for TCP sockets when they are
+/// in the connected state.
+///
+/// @par Flags
+/// This class is declares the following flags.
+///
+/// @li @b SOF_TIMESTAMPING_RX_HARDWARE:
+/// Return the original, unmodified time stamp as generated by the hardware.
+///
+/// @li @b SOF_TIMESTAMPING_RX_SOFTWARE:
+/// If SOF_TIMESTAMPING_RX_HARDWARE is off or fails, then do it in
+/// software.
+///
+/// @li @b SOF_TIMESTAMPING_RAW_HARDWARE:
+/// Return original raw hardware time stamp.
+///
+/// @li @b SOF_TIMESTAMPING_SOFTWARE:
+/// Return system time stamp generated in software.
+///
+/// @ingroup module_ntsu
 struct TimestampUtil {
-    enum {  //copied from include/linux/errqueue.h
-        e_SCM_TSTAMP_SND,
-        e_SCM_TSTAMP_SCHED,
-        e_SCM_TSTAMP_ACK
-    };
+    enum {
+        e_SCM_TSTAMP_SND = 0,
+        e_SCM_TSTAMP_SCHED = 1,
+        e_SCM_TSTAMP_ACK = 2,
 
-    enum {  //copied from include/asm-generic/socket.h
         e_SO_TIMESTAMPNS   = 35,
         e_SO_TIMESTAMPING  = 37,
         e_SCM_TIMESTAMPNS  = e_SO_TIMESTAMPNS,
-        e_SCM_TIMESTAMPING = e_SO_TIMESTAMPING
-    };
+        e_SCM_TIMESTAMPING = e_SO_TIMESTAMPING,
 
-    enum {
+        // Timestamp generation.
         e_SOF_TIMESTAMPING_TX_HARDWARE = (1 << 0),
         e_SOF_TIMESTAMPING_TX_SOFTWARE = (1 << 1),
+        e_SOF_TIMESTAMPING_TX_SCHED = (1 << 8),
+        e_SOF_TIMESTAMPING_TX_ACK = (1 << 9),
         e_SOF_TIMESTAMPING_RX_HARDWARE = (1 << 2),
         e_SOF_TIMESTAMPING_RX_SOFTWARE = (1 << 3),
-        e_SOF_TIMESTAMPING_SOFTWARE    = (1 << 4),
 
+        e_SOF_TIMESTAMPING_TX_GENERATION = 
+            e_SOF_TIMESTAMPING_TX_HARDWARE |
+            e_SOF_TIMESTAMPING_TX_SOFTWARE |
+            e_SOF_TIMESTAMPING_TX_SCHED |
+            e_SOF_TIMESTAMPING_TX_ACK,
+
+        e_SOF_TIMESTAMPING_RX_GENERATION = 
+            e_SOF_TIMESTAMPING_RX_HARDWARE |
+            e_SOF_TIMESTAMPING_RX_SOFTWARE,
+
+        // Timestamp reporting.
+        e_SOF_TIMESTAMPING_SOFTWARE    = (1 << 4),
+        e_SOF_TIMESTAMPING_SYS_HARDWARE = (1 << 5),
```

Comment:
`SOF_TIMESTAMPING_SYS_HARDWARE`  flag is deprecated afaik, and I suggest not to use it

Reply:
Done